src/main/java/com/snowflake/kafka/connector/internal/streaming/IngestionMethodConfig.java

```diff
@@ -48,6 +52,28 @@ public static String[] allIngestionTypes() {
     return result;
   }
 
+  /**
+   * Returns the Ingestion Method found in User Configuration for Snowflake Kafka Connector.
+   *
+   * <p>Default is always {@link IngestionMethodConfig#SNOWPIPE}.
+   */
+  public static IngestionMethodConfig determineIngestionMethod(Map<String, String> inputConf) {
+    if (inputConf == null
+        || inputConf.isEmpty()
+        || !inputConf.containsKey(SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT)) {
+      return IngestionMethodConfig.SNOWPIPE;
+    } else {
+      try {
+        return IngestionMethodConfig.valueOf(
+            inputConf
+                .get(SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT)
+                .toUpperCase(Locale.ROOT));
+      } catch (IllegalArgumentException ex) {
+        return IngestionMethodConfig.SNOWPIPE;
```

Comment:
Should we error out in this case? This could lead to customer confusions

Reply:
This shouldnt happen as such since we do validations on connector start but I can see this could be confusing if used elsewhere, let me throw an error here.